src/nspawn/nspawn.c

```diff
@@ -4838,6 +4856,11 @@ static int merge_settings(Settings *settings, const char *path) {
             !strv_isempty(settings->bind_user))
                 strv_free_and_replace(arg_bind_user, settings->bind_user);
 
+        if ((arg_settings_mask & SETTING_BIND_USER_SHELL) == 0 &&
```

Comment:
For newer code, please use `!FLAGS_SET(arg_setting_mask, SETTING_BIND_USER_SHELL)`